cpp/src/fil/infer.cu

```diff
@@ -576,21 +576,53 @@ __global__ void infer_k(storage_type forest, predict_params params) {
   }
 }
 
+void set_carveout(void* kernel, int footprint, int max_shm) {
+  CUDA_CHECK(
+    cudaFuncSetAttribute(kernel, cudaFuncAttributePreferredSharedMemoryCarveout,
+                         // footprint in % of max_shm, rounding up
+                         (100 * footprint + max_shm - 1) / max_shm));
+  CUDA_CHECK(cudaFuncSetAttribute(
```

Comment:
Consider the case where there are several FIL models with different parameters used in the same process on the same GPU. Will this still work?